### PR TITLE
Address several issues with friends and the friends list

### LINKED ISSUE #595: BUG: Best friends is not currently implemented
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

c89f420

### Issue Description

Best friends is not implemented.

### Reproduction Steps

1) Add someone as a friend
2) Attempt to add them as a best friend
3) Nothing happens.

### Expected Behavior

A pop up shows up to add them as a best friend.  They will now have a star next to their name and be able to super chat and visit their friends properties without moderator approval.

### Environment

All

### LINKED ISSUE #632: BUG: property friends markers dont work
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

newest

### Issue Description

yes ik

### Reproduction Steps

its being fixed

### Expected Behavior

Fixed in #630 

### Environment

all

### PR DESCRIPTION
**Pull request by EmosewaMC**
- The first issue addressed is player IDs that are stored in the friends table  
    - These are now character IDs and not their LWOOBJID counterparts to allow for not only foreign key linking, but also to keep in line with what we use elsewhere.  This does introduce a new issue that needs to be resolved in that the friends query for the leaderboards does not return quite the expected results every time

- The next issue addressed is best friends
    - To keep the PR as short as I believe it can be, the best friends feature is implemented simply so that they are correctly stored in the database and this information is correctly sent to the player to indicate such.  This PR touches nothing with chat, properties, or any other features that best friends would rely on and only implements the requests

- The correct message response codes are now sent for friend requests
    - Previously, friends were implemented pretty much to function and were missing lots of implementation.  Now the following responses will come from the server for their respective actions
        - Sending a best friend request to someone now properly updates the friends list to reflect the new status.
            - Note: Best friends have a cap of 5 players.  This can be changed with a world config.
            - Note: You can request that someone be your best friend offline and there is no request that is sent to another player.  Both players much independently send requests to the other and when both players have done so the other two players receive their respective accepted requests
         - Sending a friend request to someone who you are already a friend or best friend with now correctly displays a message that says whether or not that player is already your friend.
         - Sending a friend request to a player that doesnt exist will result in a popup reflecting such
         - If a player is not online, then a message is displayed reflecting such
         - Cancelling a friend request now sends a message to the person who sent the initial request to let them know of this
         - Sending a best friend request to someone who is already a friend now correctly displays whether or not you have already sent a request or not.

- The property viewer now correctly displays whether or not you are a friend of the player.
Fixes #595 
Fixes #632

### CODE REVIEW

**Comment by EmosewaMC** [comment]
Note to self to update names to better reflect the local information in the scope of a GM.  better names would be respondingPlayer, originatingPlayer, playerToSendFriendRequestTo and playerThatSentFriendRequest

File: dCommon/AddFriendResponseType.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#ifndef __ADDFRIENDRESPONSETYPE__H__
+#define __ADDFRIENDRESPONSETYPE__H__
+
+#include <cstdint>
+
+enum class AddFriendResponseType : uint8_t {
+    ACCEPTED = 0,
```
Comment (aronwk-aaron) [inline dCommon/AddFriendResponseType.h:9]: Inconsistent whitesp6

File: dChatServer/ChatPacketHandler.cpp [hunk 2]
```diff
@@ -91,72 +96,247 @@ void ChatPacketHandler::HandleFriendlistRequest(Packet* packet) {
...
-	stmt->execute();
-	delete stmt;
+	SendFriendResponse(requestee, requestor, serverResponseCode, isAlreadyBestFriends);
+	
+	if (serverResponseCode == AddFriendResponseType::DECLINED) SendFriendResponse(requestor, requestee, serverResponseCode, isAlreadyBestFriends);
+
+	if (serverResponseCode == AddFriendResponseType::ACCEPTED) {
+		SendFriendResponse(requestor, requestee, serverResponseCode, isAlreadyBestFriends);
+
+		// Add the goons to their friends list
```
Comment (aronwk-aaron) [inline dChatServer/ChatPacketHandler.cpp:313]: Goons

File: dChatServer/ChatPacketHandler.cpp [hunk 3]
```diff
@@ -91,72 +96,247 @@ void ChatPacketHandler::HandleFriendlistRequest(Packet* packet) {
...
+			GeneralUtils::SetBit(queryFriendID, static_cast<size_t>(eObjectBits::OBJECT_BIT_PERSISTENT));
+
+			// Since this player can either be the friend of someone else or be friends with someone else
+			// their column in the database determines what bit gets set.  When the value hits 3, they
+			// are now best friends with the other player.
+			if (queryPlayerID == requestorPlayerID) {
+				bestFriendStatus |= 1ULL << 0;
+			} else {
+				bestFriendStatus |= 1ULL << 1;
+			}
```
[the remark this answers is not in the document]
Comment (EmosewaMC) [reply dChatServer/ChatPacketHandler.cpp:201]: Addressed

File: dCommon/AddFriendResponseType.h (hunk 1 above)
Comment (EmosewaMC) [reply dCommon/AddFriendResponseType.h]: Addressed

File: dChatServer/ChatPacketHandler.cpp (hunk 2 above)
Comment (EmosewaMC) [reply dChatServer/ChatPacketHandler.cpp:312]: Addressed

**Pull request merged**